### ISSUE TITLE
looper logging

### ISSUE THREAD

**Issue by michalstolarczyk** [pepkit/looper#154]
I might need to make some changes related to the introduction of `logmuse` to `looper`. `looper` logging is not working in `caravel` now

_Originally posted by @michalstolarczyk in https://github.com/pepkit/looper/pull/153#issuecomment-480306268_

**Comment by michalstolarczyk** [comment]
Since we're calling the `looper.Executor`-like objects directly in `caravel` the logger in `main()` is not being setup. Previously (before `logmuse` introduction) I've called the `looper.setup_looper_logger` function to set the level according to user's liking.

What do you think is the best solution to this issue? 

Do we add an additional `logger` argument to the `Executor.__init__()` and then provide one created with `logmuse.setup_logger` internally in `main()`? 
That would be convenient from the `caravel` perspective

**Comment by vreuter** [comment]
From "logging is not working" it's unclear what the issue is.

**Comment by vreuter** [comment]
In general, the onus of logging setup lies with the endpoint application, so `caravel` should likely directly call `setup_logger` or `logger_via_cli`, though I'm not sure what your use case is,

**Comment by nsheff** [comment]
basically, we need to make sure that the looper logging messages are printed, when we run functions from caravel.

apparently if you just `import looper` and then run some function, the output will not be logged automatically.

**Comment by michalstolarczyk** [comment]
Yes. And that's exactly what my first intention was. But calling `logmuse.setup_logger` and then for example `looper.Runner` does not provide the `Runner` the access to the logger, which was the case in before

**Comment by michalstolarczyk** [comment]
> basically, we need to make sure that the looper logging messages are printed, when we run functions from caravel.
> 
> apparently if you just `import looper` and then run some function, the output will not be logged automatically. is that right @michalstolarczyk ?

that's right

and I don't have any idea how to resolve that apart from what I suggested in my first comment. But I guess @vreuter might have a better solution in mind

**Comment by michalstolarczyk** [comment]
And I'd great if we had control over the level and target of the logger when `looper` is imported.

In `caravel` I used to set the `looper` logger level to debug if `caravel` was launched with `-d` flag, for example

**Comment by vreuter** [comment]
Please attach an example of previous behavior and new behavior.
